Update typography guidance and restructure styles section (#3343)
alphagov/govuk-design-system: src/get-started/new-typography-scale/index.md

---
layout: layout-pane.njk

> title: Updating your service to use the new typography scale

Author: I've called this 'type scale' on the other page – it's the more commonly used phrase.

I think 'type scale' is actually shortened from 'typographic scale'. So it should probably be 'type scale' or 'typographic scale'. I personally prefer the shorter one!

Reviewer: type scale sounds like a clear way to describe it.

Author: Ok let's use type scale throughout IMO

Author: ```suggestion
title: Updating your service to use the new type scale
```

section: Get started
theme: How to guides
order: 6

> description: The GOV.UK Design System team have been working on updating our typography scale provided with GOV.UK Frontend. This new scale is now available to use and will be enforced by default in GOV.UK Frontend version 6.0.0. Here's how to use it, what's changed and what to expect when implementing it into your service.

Author: @claireashworth might have a better suggestion but I think we can clear this up a bit...

```suggestion
description: The GOV.UK Design System team have been working on updating the typography scale. This new scale is now available to use. You currently need to opt into it using a feature flag, but it will be switched on by default in GOV.UK Frontend version 6.0.0. Here's how to use it, what's changed, and what to expect when implementing it into your service.
```

Reviewer: Another potential option for this section. Also, do we want to change all instances of typography scale to type scale?

```suggestion
description: The GOV.UK Design System team has updated the GOV.UK Frontend typography scale. This new scale is an available option to use now, but will be enforced by default in GOV.UK Frontend version 6.0.0. Here's how to use it, what's changed and what to expect when implementing it into your service.
```

Author: Yeah good point, I think it should say 'type scale' everywhere

Author: ```suggestion
description: The GOV.UK Design System team has updated the GOV.UK Frontend type scale. This new scale is an available option to use now, but will be enforced by default in GOV.UK Frontend version 6.0.0. Here's how to use it, what's changed and what to expect when implementing it into your service.
```

---

{% from "govuk/components/table/macro.njk" import govukTable %}
{% from "govuk/components/warning-text/macro.njk" import govukWarningText %}

> The GOV.UK Design System team have been working on updating our typography scale provided with GOV.UK Frontend. This new scale is now available to use and will be enforced by default in GOV.UK Frontend version 6.0.0. Here's how to use it, what's changed and what to expect when implementing it into your service.

Author: (Same as the comment on line 7)

Reviewer: Ditto depending on the option we pick for use

Author: ```suggestion
The GOV.UK Design System team has updated the GOV.UK Frontend type scale. This new scale is an available option to use now, but will be enforced by default in GOV.UK Frontend version 6.0.0. Here's how to use it, what's changed and what to expect when implementing it into your service.
```

## Changes we've made
The new typography scale includes the following changes from the previous scale:

{{ govukTable({
    caption: "Changes to the typography scale",
    captionClasses: "govuk-table__caption--m",
    head: [
        {

>             text: "Typography scale value"

Author: On the type scale page in the styles section, we call this "Point on type scale". I don't know which I prefer tbh! Any opinions?

Reviewer: I don't have an opinion on which one tbh

Author: Decision: go with 'point on type scale'!

Author: ```suggestion
            text: "Point on type scale"
```

        },
        {
            text: "Change"
        }
    ],
    rows: [
        [
            {
                text: "14"
            },
            {
                text: "Deprecated as a scale value and will be removed in GOV.UK Frontend version 6.0.0"
            }
        ],
        [
            {
                text: "16"
            },
            {
                text: "16px across all screen sizes and no longer drops to 14px on small screens"
            }
        ],
        [
            {
                text: "19"
            },
            {
                text: "19px across all screen sizes and no longer drops to 16px on small screens"
            }
        ],
        [
            {
                text: "24"
            },
            {

>                 text: "21px instead of 18px on small screens, plus line height adjustment"

Reviewer: Can I check my understanding – 24px used to drop to 18px, but now it's only dropping to 21px? If so, we might want to amend the text a little to clarify unless it's more important to stay consistent with the other text entries.

>                 text: "21px instead of 18px on small screens, plus line height adjustment"

Author: Decision: make the line height adjustments specific, maybe restructure how we're wording these

            }
        ],
        [
            {
                text: "27"
            },
            {
                text: "21px instead of 18px on small screens, plus line height adjustment"
            }
        ],
        [
            {
                text: "36"
            },
            {
                text: "27px instead of 24px on small screens, plus line height adjustment"
            }
        ]
    ]
}) }}

[Read more about this work and why we're updating our typography scale](https://designnotes.blog.gov.uk/2022/12/12/making-the-gov-uk-frontend-typography-scale-more-accessible/).

## How to use the new typography scale
You can opt into the new scale by setting the feature flag variable `$govuk-new-typography-scale` at the root of your service's Sass to `true`:

```scss
$govuk-new-typography-scale: true;
```

> This will automatically use the new typography scale. You don't need to change any of your Sass to use the new scale.

Reviewer: ```suggestion
This will automatically use the new typography scale. You do not need to change any of your Sass to use the new scale.
```

{{ govukWarningText({

>   text: "Please be aware that this is an early release of the new typography scale. We may release significant changes to the scale in releases between now and GOV.UK Frontend 6.0.0. Please read version release notes to see if you are affected."

Author: I'm not sure if we say please lol. One for a content designer to look at!

>   text: "Please be aware that this is an early release of the new typography scale. We may release significant changes to the scale in releases between now and GOV.UK Frontend 6.0.0. Please read version release notes to see if you are affected."

Reviewer: ```suggestion
  text: "Please be aware this is an early release of the new typography scale. We might make significant changes to the scale in releases from now until GOV.UK Frontend 6.0.0. Please read version release notes to see if you are affected."
```

}) }}

## Testing your service against the new typography scale

> If your service is using bespoke elements made using GOV.UK Frontend eg: bespoke components or specific page templates, you should test your service against the new typography scale to assess if any adjustments are required. Some key things to look for are:

Author: ```suggestion
If your service uses custom elements made using GOV.UK Frontend, test your service against the new typography scale to assess if you need to make any adjustments. Some key things to look for are:
```

> - Spacing. Your bespoke elements may now appear squashed with increased font sizes on small screens.

Author: ```suggestion
- Spacing. With the increased font sizes, your custom elements may now look squashed on small screens.
```

Reviewer: to follow the style guide exactly (which isn't always possible) we try to avoid using bold, the bullet's should start lower case unless they're a name of a thing and they should be one sentence that does not have a full stop. I often use dashes as separators between the thing and the info about the thing

Author: Decision: use bullets and dashes

Author: ```suggestion
- spacing – your bespoke elements may now appear squashed with increased font sizes on small screens
```

> - Specific positioning values. For example an absolutely positioned element. These values may need to be adjusted 

Author: ```suggestion
- Specific positioning values. You may need to adjust these values, for example an `absolute` positioned element. 
```

Author: ```suggestion
- specific positioning values – you may need to adjust these values, for example an `absolute` positioned element 
```
